assets/tests/components/routePicker.test.tsx

```diff
@@ -2,85 +2,120 @@ import { mount } from "enzyme"
-  const tree = renderer
-    .create(<RoutePicker routes={null} selectedRouteIds={[]} />)
-    .toJSON()
-
-  expect(tree).toMatchSnapshot()
-})
-
-test("clicking a route selects it", () => {
-  const mockDispatch = jest.fn()
-  const routes = [{ id: "id" }]
-
-  const routePicker = mount(
-    <DispatchProvider dispatch={mockDispatch}>
+const mockRouteFilter: RouteFilter = {
+  filterType: "id",
+  filterText: "",
+  handleTypeChange: jest.fn(),
+  handleTextInput: jest.fn(),
+  clearTextInput: jest.fn(),
+}
+jest.fn(() => mockRouteFilter)
+jest.mock("../../src/hooks/useRouteFilter", () => ({
+  __esModule: true,
+  default: () => mockRouteFilter,
+  filterRoutes: jest.fn(routes => routes),
+}))
+
+describe("RoutePicker", () => {
+  test("renders a list of routes", () => {
+    const routes: Route[] = [
+      { id: "28" },
+      { id: "39" },
+      { id: "71" },
+      { id: "73" },
+      { id: "111" },
+    ]
+    const selectedRouteIds: RouteId[] = ["28", "39"]
+
+    const tree = renderer
+      .create(
+        <RoutePicker routes={routes} selectedRouteIds={selectedRouteIds} />
+      )
+      .toJSON()
+
+    expect(tree).toMatchSnapshot()
+  })
+
+  test("renders a loading/empty state", () => {
+    const tree = renderer
+      .create(<RoutePicker routes={null} selectedRouteIds={[]} />)
+      .toJSON()
+
+    expect(tree).toMatchSnapshot()
+  })
+
+  test("clicking a route selects it", () => {
+    const mockDispatch = jest.fn()
+    const routes = [{ id: "id" }]
+
+    const routePicker = mount(
+      <DispatchProvider dispatch={mockDispatch}>
+        <RoutePicker routes={routes} selectedRouteIds={[]} />
+      </DispatchProvider>
+    )
+
+    routePicker
+      .find(".m-route-picker__route-list-button")
+      .first()
+      .simulate("click")
+
+    expect(mockDispatch).toHaveBeenCalledWith(selectRoute("id"))
+  })
+
+  test("clicking a selected route deselects it", () => {
+    const mockDispatch = jest.fn()
+    const routes = [{ id: "id" }]
+
+    const routePicker = mount(
+      <DispatchProvider dispatch={mockDispatch}>
+        <RoutePicker routes={routes} selectedRouteIds={["id"]} />
+      </DispatchProvider>
+    )
+
+    routePicker
+      .find(".m-route-picker__route-list-button--selected")
+      .first()
+      .simulate("click")
+
+    expect(mockDispatch).toHaveBeenCalledWith(deselectRoute("id"))
+  })
+
+  test("clicking in the list of selected routes deselects a route", () => {
+    const mockDispatch = jest.fn()
+    const routes = [{ id: "id" }]
+
+    const routePicker = mount(
+      <DispatchProvider dispatch={mockDispatch}>
+        <RoutePicker routes={routes} selectedRouteIds={["id"]} />
+      </DispatchProvider>
+    )
+
+    routePicker
+      .find(".m-route-picker__selected-routes-button")
+      .first()
+      .simulate("click")
+
+    expect(mockDispatch).toHaveBeenCalledWith(deselectRoute("id"))
+  })
+
+  test("inputting filter text updates the route filter", () => {
```

Comment:
There's more functionality that's not tested yet: Changing types, and clearing the input.

Reply:
Changing types is waiting til there is a second value to test changing to. Added a test for clearing the input.